src/IO/H5/Helpers.cpp

```diff
@@ -339,24 +442,64 @@ bool contains_attribute(const hid_t file_id, const std::string& group_name,
          std::end(names);
 }
 
-DataVector read_data(const hid_t group_id, const std::string& dataset_name) {
-  // Read a DataVector from the group
-  const hid_t dataset_id =
-      H5Dopen2(group_id, dataset_name.c_str(), h5p_default());
-  CHECK_H5(dataset_id, "could not open dataset '" << dataset_name << "'");
-  // Get the number of points. These do not need a "close" call.
-  const hid_t space_id = H5Dget_space(dataset_id);
-  CHECK_H5(space_id, "Failed to open dataspace");
-  const hid_t number_of_points = H5Sget_simple_extent_npoints(space_id);
-  CHECK_H5(number_of_points, "Failed to get number of points");
-  H5Sclose(space_id);
-  // Load the data
-  DataVector data(static_cast<size_t>(number_of_points));
-  CHECK_H5(H5Dread(dataset_id, h5_type<double>(), h5s_all(), h5s_all(),
-                   h5p_default(), static_cast<void*>(data.data())),
-           "Failed to read data");
-  CHECK_H5(H5Dclose(dataset_id), "Failed to close dataset");
-  return data;
+hid_t open_dataset(const hid_t group_id,
+                   const std::string& dataset_name) noexcept {
+  const hid_t dataset_id = H5Dopen2(
+                           group_id, dataset_name.c_str(), h5p_default());
+  CHECK_H5(dataset_id, "Failed to open dataset '" << dataset_name << "'");
+  return dataset_id;
+}
+
+void close_dataset(const hid_t dataset_id) noexcept {
+CHECK_H5(H5Dclose(dataset_id), "Failed to close dataset");
```

Comment:
Please run `git clang-format -f` before you commit in the future.

Reply:
Not a good idea because I then end up with dozens of whitespace changes that have to be individually dealt with. A better idea is to do clang-format -i -lines=x:y but every once in a while I forget to do it on something I change

Reply:
It is a good idea. It formats only the lines you touched, which may need whitespace changes.

Reply:
Ok, in that case I agree

Reply:
You still need to format this line properly.